src/main/java/io/r2dbc/mssql/MssqlConnectionConfiguration.java

```diff
@@ -100,6 +100,38 @@ public static Builder builder() {
         return new Builder();
     }
 
+    /**
+     * Create a new {@link Builder} and configure it with this the values of this configuration instance.
+     *
+     * @return a {@link Builder} with this configuration
+     */
+    public Builder toBuilder() {
```

Comment:
Instead of having a builder method, we could expose a package-private method `MssqlConnectionConfiguration withServer(…)` that returns just a new instance of `MssqlConnectionConfiguration` with hostname/port applied.